### ISSUE TITLE
HttpRepl doesn't respect servers:url as a base automatically

### ISSUE THREAD

**Issue by shanselman** [dotnet/HttpRepl#148]
I've got a simple node app with a swagger.json like this.
[swagger.txt](https://github.com/aspnet/HttpRepl/files/3474294/swagger.txt)

`{
  "openapi": "3.0.0",
  "servers": [
    {
      "url": "/api/v1"
    }
  ]...`

And while it is correct and valid swagger, it's not automatically usable with httpRepl without manual and non-intuitive base url setting.

`set base https://MYSITE/api/v1
set swagger https://MYSITE/swagger.json`

The base setting should be implied by the single `servers:url` and this should work without manual set base or manual set swagger.

Second ask - Would be nice to "Set swagger LOCALFILE" as well. Or even "`dotnet watch httprep --swagger LOCALFILE`" for iterative development. 

cc @tlmii

**Comment by tlmii** [comment]
The OpenAPI 3.0 spec info for the servers array can be found [here](https://swagger.io/docs/specification/api-host-and-base-path/) (examples in YAML only for whatever reason). But in general, it just looks like:

```json
{
  "openapi": "3.0.0",
  "servers": [
    {
      "url": "[relative or absolute url to the base address]",
      "description": "[optional markdown formatted description of the server's use case]"
    },
    {
      "url": "...",
      "description": "..."
    }
  ]
}
```

The 2.0 swagger spec had a more simplistic way of dealing with this that only supported a single base address. But it still allowed it to be specified. You can see some detail on that [here](https://swagger.io/docs/specification/2-0/api-host-and-base-path/). But again, generally, it looks like this:

```json
{
  "swagger": "2.0",
  "host": "example.com",
  "basePath": "api/v1",
  "schemes": [
    "https",
    "http"
  ]
}
```

The current implementation ignores both of these. Not only does this force the user to do redundant work in many cases (specifying the base address and swagger location when the latter already specifies the former), but in cases like Scott's it can be pretty opaque as to why it doesn't work. 

The way the tool is structured right now, it seems to be geared heavily towards the base address and extrapolating from that. But since the swagger file can specify a base address, it makes me wonder if that isn't backwards.

For instance, right now, the only (non-help) command line parameter for the tool is to pass it the base address:

`dotnet httprepl https://example.com/`

That essentially just calls `set base https://example.com/` immediately after the tool starts, which in turn, if the swagger file can be located, might also set the swagger file.

It seems like a better implementation might be to allow you to specify the base address OR the swagger address, and then also implement understanding of the above information that the spec can provide. So you could do, for instance:

`dotnet httprepl --set-base https://example.com/`
or
`dotnet httprepl --set-swagger https://example.com/swagger.json/`

and the tool will figure out what it can in either case.

Some open questions:

- How to handle the multiple servers in the OpenAPI 3 spec? An easy first step would just be to use the first one, and we can provide a way to configure it later. 
- How to handle the base address changing via the swagger file? For instance, if they `set base https://example.com/` and we auto-detect the swagger file at /swagger.json, and that swagger file says the base address should be `https://example.com/api/v1/`, do we automatically switch? Or do we leave it since the user explicitly set the base?

cc @jimmylewis @jodavis @bhsubra @metoly

**Comment by tlmii** [comment]
Closing this issue since it was addressed with #179
